optimum/neuron/modeling_seq2seq.py

```diff
@@ -0,0 +1,912 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""NeuroModelForXXX classes for seq2seq models' inference on neuron devices."""
+import copy
+import logging
+import os
+import shutil
+from abc import abstractmethod
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+from huggingface_hub import snapshot_download
+from transformers import AutoConfig, AutoModelForSeq2SeqLM, GenerationConfig
+from transformers.generation.beam_search import BeamScorer
+from transformers.generation.logits_process import (
+    LogitsProcessorList,
+)
+from transformers.generation.stopping_criteria import (
+    MaxLengthCriteria,
+    MaxTimeCriteria,
+    StoppingCriteriaList,
+)
+from transformers.generation.utils import (
+    BeamSearchOutput,
+    GreedySearchOutput,
+)
+from transformers.modeling_outputs import Seq2SeqLMOutput
+
+from ..exporters.neuron import (
+    NeuronConfig,
+    main_export,
+)
+from ..exporters.neuron.model_configs import *  # noqa: F403
```

Comment:
Not a big fan of this.

Reply:
removed, it's not necessary here anymore. Trace left during the dev, sorry for that.